### PR TITLE
Pin dependency versions in Dockerfile.tpu 

### PR DESCRIPTION
**Pull request by richardsliu**
Pin the dependencies in Dockerfile.tpu to versions that work for Trillium

### CODE REVIEW

**Comment by bvrockwell** [comment]
Hi @youkaichao it looks like we have a failing tpu test because of depyf not supporting torch nightly (breaking change with check_fn). Any tips on what to do?

**Comment by youkaichao** [comment]
thanks for the heads-up! let me fix it really quickly.

File: Dockerfile.tpu [hunk 1]
```diff
@@ -9,11 +9,13 @@ RUN apt-get update && apt-get install -y \
     git \
     ffmpeg libsm6 libxext6 libgl1
 
-# Install the TPU and Pallas dependencies.
-RUN --mount=type=cache,target=/root/.cache/pip \
-    python3 -m pip install torch_xla[tpu] -f https://storage.googleapis.com/libtpu-releases/index.html
-RUN --mount=type=cache,target=/root/.cache/pip \
-    python3 -m pip install torch_xla[pallas] -f https://storage.googleapis.com/jax-releases/jax_nightly_releases.html -f https://storage.googleapis.com/jax-releases/jaxlib_nightly_releases.html
+# Install PyTorch and PyTorch XLA.
+RUN pip install --pre torch==2.6.0.dev20241028+cpu torchvision==0.20.0.dev20241028+cpu --index-url https://download.pytorch.org/whl/nightly/cpu
```
Comment (youkaichao) [inline Dockerfile.tpu:13]: why do we need to install a cpu version first?

**Comment by youkaichao** [comment]
@bvrockwell the `depyf` issue should be solved by https://github.com/thuml/depyf/pull/64 . I think there's still an torch xla error from the ci results.

**Comment by DarkLight1337** [comment]
Closing as superseded by #10010

**Pull request closed without merging**